extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/ServletProducer.java

```diff
@@ -22,4 +23,10 @@ HttpServletRequest request() {
     HttpServletResponse response() {
         return (HttpServletResponse) ServletRequestContext.requireCurrent().getServletResponse();
     }
+
+    @Produces
+    @RequestScoped
+    HttpSession session() {
+        return ((HttpServletRequest) ServletRequestContext.requireCurrent().getServletRequest()).getSession();
```

Comment:
In fact, you can simplify this to `return request().getSession()` but the current version is ok too ;-).

Reply:
Indeed. I'll update this, too.